ThreeDSecure/src/main/java/com/braintreepayments/api/ThreeDSecureActivity.java

```diff
@@ -16,10 +16,13 @@
  */
 public class ThreeDSecureActivity extends AppCompatActivity implements CardinalValidateReceiver {
 
+    static final String EXTRA_ERROR_MESSAGE = "com.braintreepayments.api.ThreeDSecureActivity.EXTRA_ERROR_MESSAGE";
     static final String EXTRA_THREE_D_SECURE_RESULT = "com.braintreepayments.api.ThreeDSecureActivity.EXTRA_THREE_D_SECURE_RESULT";
     static final String EXTRA_VALIDATION_RESPONSE = "com.braintreepayments.api.ThreeDSecureActivity.EXTRA_VALIDATION_RESPONSE";
     static final String EXTRA_JWT = "com.braintreepayments.api.ThreeDSecureActivity.EXTRA_JWT";
 
+    static final int RESULT_COULD_NOT_START_CARDINAL = RESULT_FIRST_USER;
```

Comment:
Did the Cardinal docs specify that `RESULT_FIRST_USER` (1) is an error constant? Just curious where this comes from.

Reply:
Yeah it's an [Android constant](https://developer.android.com/reference/android/app/Activity#RESULT_FIRST_USER). Any error code >= 1 is a considered a custom error. It helps to differentiate from `Activity.RESULT_CANCELED` that gets sent by the OS when the device's back button is pressed.

Reply:
Gotcha, it also looks like prefixing with `RESULT` conforms to Android conventions for these constants right?

Reply:
Also why set it to 1 (RESULT_FIRST_USER) and not some other custom int value?

Reply:
Yeah there isn't a hardcore convention i'm aware of but leading with `RESULT_` feels expressive in this case. It could be another value as long as it isn't `-1 (RESULT_OK)` or `0 (RESULT_CANCELED)`.

I've seen some start with `RESULT_FIRST_USER` or even `RESULT_FIRST_USER + 1`. As long as it doesn't conflict with system return codes though it could be any value greater than 0.